src/ovp_pipeline/commands/digest_handler.py

```
@@ -239,23 +264,107 @@ def _today_utc_date() -> str:
+    deduped_slugs = [
+        slug for slug in dict.fromkeys(evergreen_slugs) if slug
+    ]
+
+    if not deduped_crystals and not deduped_slugs:
+        return ""
+
+    parts: list[str] = ["\n## Sources\n"]
+    if deduped_crystals:
+        parts.append("**Crystals**")
+        for link in deduped_crystals:
+            parts.append(f"- {link}")
```

Ruff (0.15.12)

[warning] Use `list.extend` to create a transformed list

(PERF401)